packages/orbit-components/src/hooks/usePrevious/index.js

```diff
@@ -0,0 +1,14 @@
+// @flow
+import * as React from "react";
+
+const usePrevious = (value: any) => {
+  const ref = React.useRef();
+
+  React.useEffect(() => {
+    ref.current = value;
+  });
+
+  return ref.current;
```

Comment:
Seems like redundant hook, it used only once and probably it will be better just to use inside component without moving that logic to custom hook :)

Reply:
That's true, it's not used elsewhere yet. On the other hand, as the Modal component is already pretty huge, I tried to extract as much irrelevant logic as possible, to make the file a bit more readable and easier to skim through.

Does it make sense?

If you prefer it colocated with the component, I can put it there of course.